daemon/api.go

```diff
@@ -367,6 +374,59 @@ func webify(result map[string]interface{}, resource string) map[string]interface
 	return result
 }
 
+func searchStore(c *Command, r *http.Request) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("router can't find route for snaps")
+	}
+
+	query := r.URL.Query()
+
+	auther, err := c.d.auther(r)
+	if err != nil && err != auth.ErrInvalidAuth {
+		return InternalError("%v", err)
+	}
+
+	remoteRepo := newRemoteRepo()
+	found, err := remoteRepo.FindSnaps(query.Get("q"), query.Get("channel"), auther)
+	if err != nil {
+		status := http.StatusInternalServerError
+		if e, ok := err.(*store.HTTPError); ok && e != nil {
```

Comment:
That feels slightly dubious. The fact we cannot obtain the snaps the user asked for should probably be a single error code on our side, no matter what the store tells us. Let's say something breaks in the store and starts answering with 404. That still shouldn't be a 404 on our end.